nixos/tests/haproxy.nix

```diff
@@ -43,5 +47,13 @@ import ./make-test-python.nix ({ pkgs, ...}: {
     assert "haproxy_process_pool_allocated_bytes" in machine.succeed(
         "curl -k http://localhost:80/metrics"
     )
+
+    with subtest("reload"):
+        machine.succeed("systemctl reload haproxy")
+
+    with subtest("seamless reload"):
+        machine.systemctl("start haproxy-reloader")
+        machine.succeed("echo http://localhost:80/index.txt | http-getter -i1 -n8 -l1")
```

Comment:
I am not sure this test is solid enough. Depending on the test machine's performance, the reloading might not complete before `http-getter` stops sending requests, so it may not even be tested. If necessary I can write a proper multi-threaded test  that ensures that we keep sending requests until haproxy has finished reloading, and that HTTP connections opened before the reload are still open afterwards.

Reply:
https://www.haproxy.com/de/blog/truly-seamless-reloads-with-haproxy-no-more-hacks/ mentions a quite sophisticated load generator setup to reproduce these issues, and only on a very small percentage.

I doubt this will uncover these issues - maybe just change the test to do a simple reload, and ensure it still replies afterwards.

Reply:
Right, I will change the test to ensure that it's still replying afterwards.